### PR TITLE
Add merge option to CLI. It will update or add only provided paths.

### PR DESCRIPTION
**Pull request by MarcinMaciaszek**
There is a need to add this option to CLI in fact that when you are trying to push an updated library with a single path it removes others.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,9 @@ rfhub2-cli --no-installed-keywords ../your_repo ../your_other_repo
 ``` rfhub2-cli --load-mode=append ../your_repo ../your_other_repo```
 - `update`, which will compare existing collections with newly found ones, and update existing, remove obsolete and add new ones  
 ``` rfhub2-cli --load-mode=update ../your_repo ../your_other_repo```
+- `merge`, which will compare existing collections with newly found ones, and update existing, remove old versions and add new ones <br>
```
Comment (MaciejWiczk) [inline README.md]: I would rephrase that a bit:
`merge` - updates only matched collections, does nothing with not matched ones.

File: rfhub2/cli/keywords/keywords_importer.py [hunk 2]
```diff
@@ -77,7 +77,12 @@ def import_libraries(self) -> Tuple[int, int]:
             loaded_collections = self.update_collections(
                 existing_collections, collections
             )
-            self.delete_outdated_collections(existing_collections, collections)
+            if self.load_mode == "merge":
+                self.delete_outdated_collections(
+                    existing_collections, collections, obsolete=False
```
Comment (MaciejWiczk) [inline rfhub2/cli/keywords/keywords_importer.py]: maybe remove_obsolete or remove not_matched? What do You think?

File: tests/acceptance/resources/variables.resource [hunk 3]
```diff
@@ -34,7 +34,10 @@ ${hamburger}                       //*[@id="root"]/div/header/div/button/span[1]
 ${FIXTURES_PATH}                   ${CURDIR}${/}..${/}..${/}fixtures
 ${INITIAL_FIXTURES}                ${FIXTURES_PATH}/initial/
 ${BACKUP_FIXTURES}                 ${FIXTURES_PATH}/initial_bkp/
+${BACKUP_MERGED_FIXTURES}          ${FIXTURES_PATH}/merged_bkp/
```
Comment (MarcinMaciaszek) [inline tests/acceptance/resources/variables.resource]: This line should be removed

File: rfhub2/cli/keywords/keywords_importer.py [hunk 4]
```diff
@@ -103,11 +108,21 @@ def delete_outdated_collections(
...
         existing_collections: List[Collection],
         new_collections: List[CollectionUpdateWithKeywords],
+        obsolete: bool = True,
     ) -> Set[int]:
-        """Deletes outdated collections"""
+        """Deletes outdated collections
+        :param existing_collections: List of existing collections
+        :param new_collections: List of new collections
+        :param obsolete: removes obsolete collections as well
+        :return List of deleted collections
```
Comment (MaciejWiczk) [inline rfhub2/cli/keywords/keywords_importer.py]: Set of deleted collection_ids

File: tests/acceptance/resources/keywords.resource [hunk 5]
```diff
@@ -67,6 +67,7 @@ Verify URL Is Reachable
 
 ${endpoint} Endpoint Should Have ${n} Items
     ${response}    Get Request For Endpoint    endpoint=${endpoint}
+    Log List    ${response}
```
Comment (MarcinMaciaszek) [inline tests/acceptance/resources/keywords.resource]: This line should be removed

File: tests/acceptance/cli_population.robot [hunk 6]
```diff
@@ -103,6 +103,43 @@ Running Cli Update Load Mode Second Time Should Leave Collections Untouched
     ...    Run Cli Package With Options
     ...    --mode=insert --no-installed-keywords
 
+Cli Merge Load Mode Should Update Existing Libraries And Do Not Remove Not Provided Paths
+    [Documentation]     Cli Merge Load Mode Should Leave Application
```
Comment (MaciejWiczk) [inline tests/acceptance/cli_population.robot:107]:
```
[Documentation]    Cli Merge Load Mode Should Leave Application
    ...    With Matched Collections Updated Without Not Matched Collection Removal

File: tests/acceptance/cli_options.robot [hunk 7]
```diff
@@ -61,7 +61,7 @@ Documentation For Load Mode Should Be Displayed Properly
     [Documentation]    Documentation For Load Mode Should Be Displayed Properly
     [Tags]    rfhub2-64    update
     Output Should Contain
-    ...    -l, --load-mode [insert|append|update]
+    ...    -l, --load-mode [insert|append|update|merge]
```
Comment (MaciejWiczk) [inline tests/acceptance/cli_options.robot:64]: rfhub2-330 tag to tags to connect it with this PR

File: tests/acceptance/resources/keywords.resource (hunk 5 above)
Comment (MaciejWiczk) [inline tests/acceptance/resources/keywords.resource]: Needed?

File: README.md (hunk 1 above)
Comment (MarcinMaciaszek) [reply README.md]: fixed

File: rfhub2/cli/keywords/keywords_importer.py (hunk 2 above)
Comment (MarcinMaciaszek) [reply rfhub2/cli/keywords/keywords_importer.py]: fixed

File: rfhub2/cli/keywords/keywords_importer.py (hunk 4 above)
Comment (MarcinMaciaszek) [reply rfhub2/cli/keywords/keywords_importer.py]: fixed

File: rfhub2/templates/index.html [hunk 8]
```diff
@@ -1 +1 @@
...
...1:0},c=[];function f(e){if(t[e])return t[e].exports;var r=t[e]={i:e,l:!1,exports:{}};return i[e].call(r.exports,r,r.exports,f),r.l=!0,r.exports}f.m=i,f.c=t,f.d=function(e,r,t){f.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},f.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},f.t=function(r,e){if(1&e&&(r=f(r)),8&e)return r;if(4&e&&"object"==typeof r&&r&&r.__esModule)return r;var t=Object.create(null);if(f.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:r}),2&e&&"string"!=typeof r)for(var n in r)f.d(t,n,function(e){return r[e]}.bind(null,n));return t},f.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return f.d(r,"a",r),r},f.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},f.p="/static/";var r=window.webpackJsonpfrontend=window.webpackJsonpfrontend||[],n=r.push.bind(r);r.push=e,r=r.slice();for(var o=0;o<r.length;o++)e(r[o]);var s=n;a()}([])</script><script src="/static/static/js/2.265ff835.chunk.js"></script><script src="/static/static/js/main.06428230.chunk.js"></script></body></html>
```
Comment (MaciejWiczk) [inline rfhub2/templates/index.html]: please checkout this one, it's a leftover from build_ui.sh, for local frontend testing

File: rfhub2/cli/keywords/keywords_importer.py (hunk 2 above)
Comment (MaciejWiczk) [reply rfhub2/cli/keywords/keywords_importer.py]: how about remove_not_matched? It's longer, but I think it will tell everything about this flag.

**Comment by MaciejWiczk** [comment]
Last thing to do is to write some docs for [Readthedocs](https://rfhub2.readthedocs.io/en/latest/). 
[Here ](https://github.com/pbylicki/rfhub2/blob/master/docs/source/run.rst#rfhub2-cli-for-keywords-documentation-can-be-run-in-three-load-modes)is a file to be changed. Also paragraph [here](https://github.com/pbylicki/rfhub2/blob/master/docs/source/run.rst#full-list-of-rfhub2-cli-options)

**Review by MaciejWiczk — APPROVED** [review]
Awesome work!

**Review by pbylicki — APPROVED** [review]
